streams/src/main/java/org/apache/kafka/streams/kstream/Repartitioned.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.kstream;
+
+import org.apache.kafka.clients.producer.internals.DefaultPartitioner;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.streams.kstream.internals.WindowedSerializer;
+import org.apache.kafka.streams.kstream.internals.WindowedStreamPartitioner;
+import org.apache.kafka.streams.processor.StreamPartitioner;
+
+/**
+ * This class is used to provide the optional parameters for internal repartition topics.
+ *
+ * @param <K> key type
+ * @param <V> value type
+ * @see KStream#repartition()
+ * @see KStream#repartition(Repartitioned)
+ */
+public class Repartitioned<K, V> implements NamedOperation<Repartitioned<K, V>> {
+
+    protected final String name;
+    protected final Serde<K> keySerde;
+    protected final Serde<V> valueSerde;
+    protected final Integer numberOfPartitions;
+    protected final StreamPartitioner<K, V> partitioner;
+
+    private Repartitioned(final String name,
+                          final Serde<K> keySerde,
+                          final Serde<V> valueSerde,
+                          final Integer numberOfPartitions,
+                          final StreamPartitioner<K, V> partitioner) {
+        this.name = name;
+        this.keySerde = keySerde;
+        this.valueSerde = valueSerde;
+        this.numberOfPartitions = numberOfPartitions;
+        this.partitioner = partitioner;
+    }
+
+    protected Repartitioned(final Repartitioned<K, V> repartitioned) {
+        this(
+            repartitioned.name,
+            repartitioned.keySerde,
+            repartitioned.valueSerde,
+            repartitioned.numberOfPartitions,
+            repartitioned.partitioner
+        );
+    }
+
+    /**
+     * Create a {@code Repartitioned} instance with the provided name used as part of the repartition topic.
+     *
+     * @param name the name used as a processor named and part of the repartition topic name.
+     * @param <K>  key type
+     * @param <V>  value type
+     * @return A new {@code Repartitioned} instance configured with processor name and repartition topic name
+     * @see KStream#repartition(Repartitioned)
```

Comment:
I've followed same standard as other configurations classes (Produced, Grouped, etc). To keep things consistent maybe worth cleaning up all the config classes with redundant `@see` tags? (in the follow up pr maybe) wdyt?

Reply:
Yeah. Was just a general inquire and we don't really have a guideline for it... If you are interested, it would be great to draft some guidelines (maybe just for Kafka Streams first, and we could propose them for other client APIs, later) as a wiki page and we could discuss them on the dev mailing list?